### PR TITLE
VZ-10529: Self-signed CA Certificate Expiry Causes Upgrade to Hang

### PR DESCRIPTION
**Pull request by mgianatagh**
Provide a summary of the change and which issue (i.e. ticket) is fixed.
If there are any dependencies, for example on a PR in another repository, list those.

### CODE REVIEW

File: platform-operator/controllers/secrets/secrets_controller.go [hunk 1]
```diff
@@ -49,40 +52,68 @@ func (r *VerrazzanoSecretsReconciler) Reconcile(ctx context.Context, req ctrl.Re
...
+		if err = r.renewClusterIssuerCertificates(); err != nil {
+			zap.S().Errorf("Failed to new all certificates issued by ClusterIssuer %s: %s", vzconst.VerrazzanoClusterIssuerName, err.Error())
 			return newRequeueWithDelay(), err
 		}
-		return res, nil
+		return r.reconcileVerrazzanoTLS(ctx, req.NamespacedName, corev1.TLSCertKey)
 	}
 
-	return ctrl.Result{}, nil
+	// Ingress secret was updated, or if there's a CA crt update the verrazzano-tls-ca copy; this will trigger
```
Comment (michael-cico) [inline platform-operator/controllers/secrets/secrets_controller.go:84]: Not sure why we need to look at the ingress secret still?  I thought we would look at either the ClusterIssuer secrets (like above) or the `verrazzano-tls-ca` cert to update downstream copies.  i.e., we were just replacing the ingress leaf cert check with the source CA secrets managed by Cert-Manager.

File: platform-operator/controllers/verrazzano/component/certmanager/issuer/cluster_issuer.go [hunk 2]
```diff
@@ -283,8 +297,8 @@ func getCertIssuerCommonName(currentCert certv1.Certificate) (string, error) {
 	return certIssuerCN, nil
 }
 
-// renewCertificate Requests a new certificate by updating the status of the Certificate object to "Issuing"
-func renewCertificate(ctx context.Context, cmclientv1 certv1client.CertmanagerV1Interface, log vzlog.VerrazzanoLogger, updateCert *certv1.Certificate) error {
+// RenewCertificate Requests a new certificate by updating the status of the Certificate object to "Issuing"
+func RenewCertificate(ctx context.Context, cmclientv1 certv1client.CertmanagerV1Interface, log vzlog.VerrazzanoLogger, updateCert *certv1.Certificate) error {
```
Comment (michael-cico) [inline platform-operator/controllers/verrazzano/component/certmanager/issuer/cluster_issuer.go:301]: Should this be refactored into the `pkg/certs` package?

File: platform-operator/controllers/secrets/secrets_controller.go [hunk 3]
```diff
@@ -49,40 +52,68 @@ func (r *VerrazzanoSecretsReconciler) Reconcile(ctx context.Context, req ctrl.Re
...
+	// a reconcile which will update any upstream copies
+	// - Cert-Manager rotates the CA cert in the self-signed/custom CA case causing it to be updated in leaf cert secret,
+	//   and we update the copy in the verrazzano-system/verrazzano-tls-ca secret
+	// - the ClusterIssuerComponent updates the verrazzano-system/verrazzano-tls-ca secret
+	letsEncrypt, err := clusterIssuer.IsLetsEncryptIssuer()
+	if err != nil {
+		zap.S().Errorf("Failed to determine if Let's Encrypt certificates are configured: %s", err.Error())
+		return newRequeueWithDelay(), err
+	}
+	if letsEncrypt && isVerrazzanoIngressSecretName(req.NamespacedName) {
```
Comment (michael-cico) [inline platform-operator/controllers/secrets/secrets_controller.go:94]: Not sure why we have to check `letsEncrypt` here?

**Review by michael-cico — APPROVED** [review]

File: tools/vz/pkg/analysis/internal/util/cluster/certificates.go [hunk 4]
```diff
@@ -38,6 +38,9 @@ func AnalyzeCertificateRelatedIssues(log *zap.SugaredLogger, clusterRoot string)
 	var issueReporter = report.IssueReporter{
 		PendingIssues: make(map[string]report.Issue),
 	}
+
+	// Get the secret configured for the Verrazzano ClusterIssuer.  This will be used to check if certificates
```
Comment (desagar) [inline tools/vz/pkg/analysis/internal/util/cluster/certificates.go:42]: looks like a dangling comment

**Review by desagar — APPROVED** [review]

File: tools/vz/pkg/analysis/internal/util/cluster/certificates.go (hunk 4 above)
Comment (mgianatagh) [reply tools/vz/pkg/analysis/internal/util/cluster/certificates.go]: fixed

**Review by michael-cico — APPROVED** [review]

**Pull request merged**